### PR TITLE
[closure-deps] Allow globs in --exclude flag

### PR DESCRIPTION
**Pull request by kjin**
This change adds the ability to specify globs in closuremakedeps's --exclude flag, instead of just path prefixes.

### CODE REVIEW

File: closure-deps/bin/closuremakedeps.js [hunk 1]
```diff
@@ -52,14 +53,15 @@ function parseArgs(args) {
...
-            'Closure Library\'s deps.js file, then there is no need to ' +
-            'include closure/goog/array/array.js as an input file as well. ' +
+        description:
+            'One or more input files to calculate dependencies for. The ' +
+            'namespaces in these files will be combined with those given ' +
+            'with the --root flag to form the set of namespaces to find ' +
+            'dependencies for. If this is a file that contains repeated ' +
+            'calls to goog.addDependency, then the calls will be used as ' +
+            'forward declarations. e.g. if you include Closure Library\'s ' +
+            'deps.js file, then there is no need to include ' +
```
Comment (12wrigja) [inline closure-deps/bin/closuremakedeps.js:63]: Seems unrelated - revert?

File: closure-deps/bin/closuremakedeps.js [hunk 2]
```diff
@@ -215,7 +226,7 @@ async function main(opt_args) {
 
   const sources = new Set((args.file || []).map(resolve));
   const roots = (args.root || []).map(resolve);
-  const excluded = new Set((args.exclude || []).map(resolve));
```
Comment (12wrigja) [inline closure-deps/bin/closuremakedeps.js:218]: Doesn't this still make sense to be treated as a set? e.g. if someone specifies `--exclude=path/to/*_test.js --exclude=path/to/*_test.js` we should only process that glob once?

File: closure-deps/bin/closuremakedeps.js [hunk 3]
```diff
@@ -215,7 +226,7 @@ async function main(opt_args) {
...
+  const excluded = (args.exclude || []).map(resolve);
```
Comment (12wrigja) [inline closure-deps/bin/closuremakedeps.js:229]: This tries to resolve a string as a path relative to the cwd - this doesn't make sense if these are now globs?

File: closure-deps/bin/closuremakedeps.js [hunk 4]
```diff
@@ -129,11 +140,11 @@ function promisify(fn, ...args) {
 
 /**
  * @param {string} pathToScan
- * @param {!Set<string>} excluded
+ * @param {!Array<string>} excluded
```
Comment (12wrigja) [inline closure-deps/bin/closuremakedeps.js:143]: Nit: maybe name this excludedGlobs so it's clearer these are not file path references.

**Review by 12wrigja — CHANGES_REQUESTED** [review]

File: closure-deps/bin/closuremakedeps.js (hunk 1 above)
Comment (kjin) [reply closure-deps/bin/closuremakedeps.js:63]: Yep, trying to figure out why Travis's clang-format is not cooperating with my local one
